pkg/service/cluster/service.go

```diff
@@ -667,10 +669,27 @@ func (s *Service) ListNodes(ctx context.Context, clusterID uuid.UUID) ([]Node, e
 			if err != nil {
 				s.logger.Error(ctx, "Failed to get number of shards", "error", err)
 			}
+			ni, err := client.NodeInfo(ctx, h)
+			if err != nil {
+				return nil, errors.Wrapf(err, "node info call of %s", h)
+			}
+			promAddr := ni.PrometheusAddress
+			if temp, err := netip.ParseAddr(ni.PrometheusAddress); err == nil {
+				promAddr = temp.String()
+			}
+			if promAddr == "" || promAddr == "0.0.0.0" || promAddr == net.ParseIP("0:0:0:0:0:0:0:0").String() {
+				promAddr = h
+			}
```

Comment:
It's better to use already implemented checks like https://pkg.go.dev/net#IP.IsUnspecified

Reply:
makes sense.
updated